### DISCUSSION TITLE
Service accounts cannot invite attendees without Domain-Wide Delegation of Authority.

### DISCUSSION THREAD

**Discussion by jstcode99** [spatie/laravel-google-calendar#294] [Q&A]
I'm building a simple implementation for get link and send invitation of diferentes meetings but I have had mistake with API when create a new envent with addAttendee and addMeetLink recive next error of http:

`{
    "error": {
        "errors": [
            {
                "domain": "calendar",
                "reason": "forbiddenForServiceAccounts",
                "message": "Service accounts cannot invite attendees without Domain-Wide Delegation of Authority."
            }
        ],
        "code": 403,
        "message": "Service accounts cannot invite attendees without Domain-Wide Delegation of Authority."
    }
}`

my code: 

`

        $e =  new Event;
        $e->name = 'A new event';
        $e->startDateTime = Carbon::now();
        $e->endDateTime = Carbon::now()->addHour();
        $e->addAttendee([
            'email' => 'john@example.com',
            'name' => 'John Doe',
            'comment' => 'Lorum ipsum',
            'responseStatus' => 'needsAction',
        ]);
        $e->addAttendee(['email' => 'anotherEmail@gmail.com']);
        $e->addMeetLink(); // optionally add a google meet link to the e

        $e->save();

        // get all future es on a calendar
        $e =  Event::get();

`

config: 

oauth-credentials
`
{
    "web": {
        "client_id": "-----key----",
        "project_id": "coral-lightning-414519",
        "auth_uri": "https://accounts.google.com/o/oauth2/auth",
        "token_uri": "https://oauth2.googleapis.com/token",
        "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
        "client_secret": "----key----"
    }
}
`

oauth-token
`
{
    "access_token": "---key---"
}
`

`
{
    "type": "service_account",
    "project_id": "coral-lightning-414519",
    "private_key_id": "---key---",
    "private_key": "-----key----,
    "client_email": "partner@coral-lightning-414519.iam.gserviceaccount.com",
    "client_id": "---cliend---",
    "auth_uri": "https://accounts.google.com/o/oauth2/auth",
    "token_uri": "https://oauth2.googleapis.com/token",
    "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
    "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/partner%40coral-lightning-414519.iam.gserviceaccount.com",
    "universe_domain": "googleapis.com"
}
`
env:

`
GOOGLE_CALENDAR_ID=key
GOOGLE_CALENDAR_IMPERSONATE=partner@coral-lightning-414519.iam.gserviceaccount.com
GOOGLE_CALENDAR_AUTH_PROFILE=service_account
`

alson I was folliwing this instrunction but I dont have good result

https://stackoverflow.com/questions/60760959/google-calendar-api-service-account-error

**Comment by hossein-askari83** [comment]
hi, hope have a good day
in GOOGLE_CALENDAR_IMPERSONATE variable instead of your service account, put your actual google admin email
that works for me

**Reply by jstcode99** [reply]
thx bro, but now I have next error I gave permisions for mi own account jstorres0211@gmail.com from console https://console.cloud.google.com/iam-admin/serviceaccounts/details/ on *The resources that this service account can accessThe resources that this service account can access* it's permissions:

1. Service account token creator
2. OpenID Connect Identity Token Creator for Service Accounts
3. Owner


 am I'm forgeting something?

**Comment by nxmndr** [comment]
I feel like there are unsaid things in the various API docs. @hossein-askari83 do you mean the account for **admin.google.com** ? I don't think we have this, yet we are running a company with a hundred websites. I tried with the **console.cloud.google.com** admin and I get :

```
Google\Service\Exception  {
  "error": "unauthorized_client",
  "error_description": "Client is unauthorized to retrieve access tokens using this method, or client not authorized for any of the scopes requested."
}.
```

I have created the service account, shared the agenda with it and added its key file and calendar ID to the config file to no avail. I am pretty sure the doc is missing something.

**Reply by jstcode99** [reply]
Yes, I had to use another meet system because Calendar Meet is impossible for me resolve it
